### PR TITLE
Additional minimizers

### LINKED ISSUE #24: support for different scipy minimizers
user should be able choose between different scipy minimizers
https://docs.scipy.org/doc/scipy/reference/tutorial/optimize.html

### PR DESCRIPTION
**Pull request by chrhck**
Add support for additional scipy minimizers

closes #24

### CODE REVIEW

File: skyllh/core/minimizer.py [hunk 1]
```diff
@@ -117,6 +120,173 @@ def is_repeatable(self, status):
...
+        xmin : 1D ndarray
+            The array containing the function arguments at the function's
+            minimum.
+        fmin : float
+            The function value at its minimum.
+        status : dict
+            The status dictionary with information about the minimization
+            process. The following information are provided:
+
+            niter : int
```
Comment (martwo) [inline skyllh/core/minimizer.py]: I believe that "niter" is incorrect here. Scipy minimizer usually return "nit". That's why I added the get_niter method to the minimizer implementation interface to abstract the different content of the status dictionary.

Comment (chrhck) [reply skyllh/core/minimizer.py]: fixed

File: skyllh/core/minimizer.py [hunk 2]
```diff
@@ -117,6 +120,165 @@ def is_repeatable(self, status):
...
+
+        # Check if method allows for bounds
+        if self._method in ["L-BFGS-B", "TNC", "SLSQP"]:
+            method_supports_bounds = True
+        elif self._method == "COBYLA":
+            # COBYLA doesn't allow for bounds, but we can convert bounds
+            # to a linear constraint
+
+            constraints = []
+            for bound_num, bound in enumerate(bounds):
```
Comment (martwo) [inline skyllh/core/minimizer.py:191]:
```suggestion
            for (bound_num, bound) in enumerate(bounds):
```

File: skyllh/core/minimizer.py [hunk 3]
```diff
@@ -117,6 +120,165 @@ def is_repeatable(self, status):
...
+            Scipy OptimizeResult
+        """
+
+        method_supports_bounds = False
+
+        constraints: Optional[List[Dict[str, Any]]]
+        constraints = None
+
+        # Check if method allows for bounds
+        if self._method in ["L-BFGS-B", "TNC", "SLSQP"]:
```
Comment (martwo) [inline skyllh/core/minimizer.py:184]:
```suggestion
        if(self._method in ["L-BFGS-B", "TNC", "SLSQP"]):
```

File: skyllh/core/minimizer.py [hunk 4]
```diff
@@ -117,6 +120,165 @@ def is_repeatable(self, status):
...
+            method_supports_bounds = True
+        elif self._method == "COBYLA":
```
Comment (martwo) [inline skyllh/core/minimizer.py:186]:
```suggestion
        elif(self._method == "COBYLA"):
```

File: skyllh/core/minimizer.py [hunk 5]
```diff
@@ -117,6 +120,165 @@ def is_repeatable(self, status):
...
+                lower, upper = bound
+                lc = {"type": "ineq",
+                      "fun": lambda x, lb=lower, i=bound_num: x[i] - lb}
+                uc = {"type": "ineq",
+                      "fun": lambda x, ub=upper, i=bound_num: ub - x[i]}
+                constraints.append(lc)
+                constraints.append(uc)
+            bounds = None
+
+        if bounds is not None and not method_supports_bounds:
```
Comment (martwo) [inline skyllh/core/minimizer.py:201]:
```suggestion
        if((bounds is not None) and (not method_supports_bounds)):
```

File: skyllh/core/minimizer.py [hunk 6]
```diff
@@ -117,6 +120,165 @@ def is_repeatable(self, status):
...
+        status : dict
+            The dictionary with the status information about the minimization
+            process.
+
+        Returns
+        -------
+        converged : bool
+            The flag if the minimization has converged (True), or not (False).
+        """
+        return status["success"]
```
Comment (martwo) [inline skyllh/core/minimizer.py:258]:
```suggestion
        return bool(status["success"])
```

Comment (martwo) [reply skyllh/core/minimizer.py:258]: I thing the success key is an int, and has to be converted to a bool.

**Pull request merged**